### ISSUE TITLE
Bodyguards system

### ISSUE THREAD

**Issue by DarkTl** [XelaPy/PyTFall#52]
Which purpose is to allow players to create MC unfamiliar with battles completely and still will not be raped by one lvl1 slime in the first non scripted battle.
We already have MC's own party where you can freely add and remove characters, with the exception of MC himself, he's always there.

1) Now we assume that MC does not want to die, thus his party, his bodyguards are always close to him, no matter his class. Except special cases like 1v1 arena battle, but it's your own desision, we will never force 1v1 battles when you only have MC and no one else. Because even for battle classes, pure mage could use someone on the front line as well.

2) If you don't have a battle class, you still can increase attack, learn spells, etc. It just won't be as effective. Thus, we don't forbid even non warrior MC to fight. However, it's up to the player. He can decide if MC will be on the battlefield or not. And without him there will be only his two bodyguards. Yes, it means more difficult fights if MC is too weak and not involved in the fighting. But what else did you expected when you didn't selected a battle class or didn't trained his battle stats? At least you are a good manager and have much more gold than a pure warrior :smile: 

3) Of course if you have no bodyguards, you have to fight on your own. But first bodyguard will be obtained during initial cutscenes. We make sure that MC is more or less protected from the beginning. However, anything else is up to player. He will dismiss her and then will go and die at the arena alone? Good riddance.

**Comment by Xela00** [comment]
There should be ways for MC to avoid the battles or to end up raped if there is no backup. That's the whole point of having an open world sim environment and not a combat oriented game. We could come up with a pre-combat called screen where it would be possible to setup MCs team even without MC present from friends/employees or whatever containers we allow.

1) We could "force" 1v1 battles, for example when we split the city in quarters once the map design is decided upon 100% (which I am still not sure we have even though Gismos map is prolly the best bet) and ventures into a bad neighborhood, but that is also a choice.

2) Yeap, that can be prolly achieved with a screen. Not just more money, I have ideas on how to make Warrior class for management players even less desirable by limited content and ability to manage that content effectively.

3) I am against this being as a rule. There is no good reason for this unless we decide to make a combat oriented game, in which case we are at the very, very wrong path and 40% of very good and complicated code is not even required for such a game.

*Basically if you want to make a main story like for a combat oriented player as you seem to be doing now, just do it. We'll make sure it is understood that it is the only story path available for the next release and a none warrior has no change if completing it, which is absolutely normal. If we create a pure management classes, than those will be purely sandbox classes with no story and just direct, game mechanics + general content available to them.

**Comment by DarkTl** [comment]
I think when someone is fighting instead of you, it still counts as avoiding battle :smile: 

You see, unlike battle classes, every non battle one should have its own way to resolve the conflict. While any battle classes can just eliminate the enemy, casanova has to use his charisma somehow. merchant should try to bribe, smart character (scientist?) could try to oversmart, manager probably will be raped if there are no warriors to help, etc.

In other words, while all battle classes act in the same way with difference inside BE only, every non battle class has its own features. That is a hell to come up with and code everywhere they needed.
And the more non combat classes MC has, the more checks they will need to avoid combat instead of just fighting.

So making a party to fight instead of MC when needed is the easest way.

**Comment by DarkTl** [comment]
To clarify, it's not like I want a strictly combat oriented game. I just cannot imagine a good game with BE but without a proper final boss battle.
Just because MC cannot fight doesn't mean that the player shouldn't either. We just don't use MC in BE if needed.

**Comment by DarkTl** [comment]
Meh -_-
